### PR TITLE
only report read-write images if we have theright spirv capability

### PR DESCRIPTION
**Pull request by rjodinchr**
We need at least StorageImageReadWithoutFormat to support read write images.

### CODE REVIEW

File: src/api.cpp [hunk 1]
```diff
@@ -726,6 +726,12 @@ cl_int CLVK_API_CALL clGetDeviceInfo(cl_device_id dev,
         size_ret = sizeof(val_uint);
         break;
     case CL_DEVICE_MAX_READ_WRITE_IMAGE_ARGS:
+        if (!device->supports_read_write_images()) {
+            val_uint = 0;
+            copy_ptr = &val_uint;
+            size_ret = sizeof(val_uint);
```
Comment (kpet) [inline src/api.cpp:732]:
```suggestion
            size_ret = sizeof(val_uint);
            break;
```

Is this missing a `break`? Unless I'm missing something, `val_uint` will always be overwritten below.

File: src/api.cpp [hunk 2]
```diff
@@ -726,6 +726,12 @@ cl_int CLVK_API_CALL clGetDeviceInfo(cl_device_id dev,
...
+        }
+        // fallthrough
```
Comment (kpet) [inline src/api.cpp:734]:
```suggestion
        [[fallthrough]];
```

We're now building as C++17 so might as well :)

File: src/device.hpp [hunk 3]
```diff
@@ -298,6 +298,11 @@ struct cvk_device : public _cl_device_id,
         return devices_support_images() ? CL_TRUE : CL_FALSE;
     }
 
+    bool supports_read_write_images() const {
+        return supports_capability(
+            spv::CapabilityStorageImageReadWithoutFormat);
```
Comment (kpet) [inline src/device.hpp:303]: I think we should also require `CapabilityStorageImageWriteWithoutFormat`.

**Review by kpet — CHANGES_REQUESTED** [review]

File: src/device.hpp (hunk 3 above)
Comment (rjodinchr) [reply src/device.hpp]: I agree, I have added it

**Review by kpet — APPROVED** [review]

**Pull request merged**